### PR TITLE
feature: Replace torrents with magnet links

### PR DESCRIPTION
**Pull request by filiptrplan**
Resolves: rivenmedia/riven-frontend#5

## Description:

The backend part of rivenmedia/riven-frontend#108. I would like to ask the reviewers to check the code carefully as I'm not yet fully familiar with the backend workings of the project. Also partially solves #726 and #605

### CODE REVIEW

**Comment by dreulavelle** [comment]
https://www.conventionalcommits.org/en/v1.0.0/

Please stick with conventional commits. Can you clean up your commit history

**Comment by filiptrplan** [comment]
> https://www.conventionalcommits.org/en/v1.0.0/
> 
> Please stick with conventional commits. Can you clean up your commit history

I am sticking to conventional commits. Can you specify what needs cleaning up? I would be happy to.

**Comment by dreulavelle** [comment]
Give me a bit and I can look it over. My home internet decided to die on me lol

File: src/controllers/items.py [hunk 1]
```diff
@@ -265,6 +272,96 @@ async def remove_item(request: Request, ids: str):
...
+def add_torrent(request: Request, id: int, magnet: str):
+    torrent_id = ""
+    try:
+        torrent_id = add_torrent_magnet(magnet)
+    except Exception:
+        raise HTTPException(status_code=500, detail="Failed to add torrent.") from None
+    
+    return set_torrent_rd(request, id, torrent_id)
+
+def reset_item_to_scraped(item: MediaItem):
```
Comment (dreulavelle) [inline src/controllers/items.py:285]: theres a reset somewhere else already

**Review by dreulavelle — APPROVED** [review]

File: src/controllers/items.py (hunk 1 above)
Comment (filiptrplan) [reply src/controllers/items.py:285]: I'll fix it.

**Pull request merged**